File: sliding-window/0643-maximum-average-subarray-i/java/Solution.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        long windowSum = 0;

        for (int i = 0; i < k; i++) {
            windowSum += nums[i];
        }

        long maxSum = windowSum;

        for (int right = k; right < nums.length; right++) {
            windowSum += nums[right] - nums[right - k];
            maxSum = Math.max(maxSum, windowSum);
        }

        return (double) maxSum / k;
    }
}

File: sliding-window/0643-maximum-average-subarray-i/cpp/Solution.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        long long windowSum = 0;

        for (int i = 0; i < k; i++) {
            windowSum += nums[i];
        }

        long long maxSum = windowSum;

        for (int right = k; right < static_cast<int>(nums.size()); right++) {
            windowSum += nums[right] - nums[right - k];
            maxSum = max(maxSum, windowSum);
        }

        return static_cast<double>(maxSum) / k;
    }
};
